book-a-reading-room-visit.api/Service/Impl/AvailabilityService.cs

```diff
@@ -91,7 +95,36 @@ public async Task<List<Seat>> GetAvailabileSeatsAsync(SeatTypes seatType, DateTi
                         join booking in _context.Set<Booking>().Where(b => b.VisitStartDate == availableOn)
                         on seat.Id equals booking.SeatId into lj
                         from subseat in lj.DefaultIfEmpty()
+                        where subseat == null
                         select new Seat { Id = seat.Id, Number = seat.Number, SeatType = seat.SeatType, SeatTypeId = seat.SeatTypeId }).ToListAsync();
         }
+
+        public async Task<List<Seat>> GetAllAvailabileSeatsAsync(DateTime availableOn, int visitDuration)
```

Comment:
Rather than pass the duration if you pass the bookingtype as the parameter, it implies it's one day or two